test/support/application_fixture_helper.rb

```diff
@@ -0,0 +1,75 @@
+# typed: ignore
+# frozen_string_literal: true
+
+module ApplicationFixtureHelper
+  TEMP_FIXTURE_DIR = ROOT.join("tmp", "fixtures").to_s
+  DEFAULT_TEMPLATE = :minimal
+
+  def setup_application_fixture
+    @old_working_dir = Dir.pwd
+  end
+
+  def teardown_application_fixture
+    Dir.chdir(@old_working_dir)
+    FileUtils.remove_entry(@app_dir, true) if defined? @app_dir
+  end
+
+  def copy_template(template)
+    copy_dir("test/fixtures/#{template}")
+    Dir.chdir(app_dir)
+  end
+
+  def app_dir
+    copy_template(DEFAULT_TEMPLATE) unless defined? @app_dir
+    @app_dir
+  end
+
+  def config
+    @config ||= Packwerk::Configuration.from_path(app_dir)
+  end
+
+  def path_to(relative_path)
+    File.join(app_dir, relative_path)
+  end
+
+  def paths_to(*relative_paths)
+    relative_paths.map { |path| path_to(path) }
+  end
+
+  def merge_into_yaml_file(relative_path, hash)
+    path = path_to(relative_path)
+    FileUtils.mkpath(File.dirname(path))
+    FileUtils.touch(path)
+    data = YAML.load_file(path) || {}
+    recursive_merge!(data, hash)
+    File.open(path, "w") { |f| YAML.dump(data, f) }
+  end
+
+  def remove_app_entry(relative_path)
+    FileUtils.remove_entry(path_to(relative_path))
+  end
+
+  private
+
+  def copy_dir(path)
+    root = FileUtils.mkdir_p(fixture_path).last
+    FileUtils.cp_r("#{path}/.", root)
+    @app_dir = root
+  end
+
+  def fixture_path
+    File.join(TEMP_FIXTURE_DIR, "#{name}-#{Time.now.strftime("%Y%m%d")}")
+  end
+
+  def recursive_merge!(hash, other_hash)
+    hash.merge!(other_hash) do |_, old_value, new_value|
```

Comment:
I think it would be better to return a new value instead of mutating the data. That way it makes it easier to reason what a method does.

Reply:
If my mental compilation is correct, then I think I can achieve this by simply switching to `hash.merge` on line 65, and then rename my method to `recursive_merge` to indicate it does not affect the parameter values.

Works for me!